include/deal.II/lac/diagonal_matrix.h

```diff
@@ -0,0 +1,288 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__diagonal_matrix_h
+#define dealii__diagonal_matrix_h
+
+
+#include <deal.II/lac/vector.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * This class represents a diagonal matrix based on a vector.
+ *
+ * @author Martin Kronbichler, 2016
+ */
+template <typename VectorType=Vector<double> >
+class DiagonalMatrix : public Subscriptor
+{
+public:
+  typedef typename VectorType::value_type value_type;
+  typedef typename VectorType::size_type size_type;
+
+  /**
+   * Constructor.
+   */
+  DiagonalMatrix();
+
+  /**
+   * Initialize with a given vector by copying the content of the vector
+   * @p vec.
+   */
+  void reinit (const VectorType &vec);
+
+  /**
+   * Returns a reference to the underlying vector for manipulation of the
+   * entries on the diagonal.
+   */
+  VectorType& get_vector();
+
+  /**
+   * Returns a read-only reference to the underlying vector.
+   */
+  const VectorType& get_vector() const;
+
+  /**
+   * Number of rows of this matrix. To remember: this matrix is an <i>m x
+   * m</i>-matrix.
+   */
+  size_type m () const;
+
+  /**
+   * Number of columns of this matrix. To remember: this matrix is an <i>n x
+   * n</i>-matrix.
+   *
+   * Depending on the layout of the underlying vector, only some indices can
```

Comment:
Oh yes, that is a copy-paste artifact that should actually have gone to two comments below.